azurerm/internal/locks/lock.go

```diff
@@ -31,7 +38,10 @@ func UnlockByName(name string, resourceType string) {
 }
 
 func UnlockMultipleByName(names *[]string, resourceType string) {
-	for _, name := range *names {
-		UnlockByName(name, resourceType)
+	for i, name := range *names {
+		// at the end of every array item add its index. We need to add this sufix because it was added during the lock process
+		uniqueValue := fmt.Sprintf("%s-arrIdx-%d", name, i)
```

Comment:
And then we don't need to worry about it here, unlocking twice in a row i think should be fine?

Reply:
Actually we need to handle the change here, also, because if you try to unlock already unlocked mutex (by using the same lock string) you will get the following error (I tested this scenarion few times):
`fatal error: sync: unlock of unlocked mutex`

Reply:
cool thaks for checking that